### PR TITLE
Fix: The first part of the new Casa Case form does not save certain input on error

### LINKED ISSUE #5760: New Casa Case form not fields not saving on error
## What type(s) of user does this feature affect?
 - supervisors
 - admins
 - volunteers

## Description

The first part of the new Casa Case form does not save certain input on error.

- Log in as an admin and go to `/casa_cases/new`
- fill in all values except for court date
- submit the form
- you get an error, however, the values for "I don't know court date" and contact types all get cleared

Image

Make sure that those values are not cleared if there is an error in the form.

## How to access the QA site
_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in [slack](https://join.slack.com/t/rubyforgood/shared_invite/zt-21pyz2ab8-H6JgQfGGI0Ab6MfNOZRIQA) #casa channel to ask questions quickly. And [discord](https://discord.gg/qJcw2RZH8Q) for office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by iamronakgupta**
### What github issue is this PR for, if any?
Resolves #5760

### What changed, and _why_?
 Fixed values for "I don't know court date" and contact types all get cleared when there is error in form

### How is this **tested**? (please write tests!) 💖💪
_Note: if you see a flake in your test build in github actions, please post in slack #casa "Flaky test: <link to failed build>" :) 💪_
_Note: We love [capybara](https://rubydoc.info/github/teamcapybara/capybara) tests! If you are writing both haml/js and ruby, please try to test your work with tests at every level including system tests like https://github.com/rubyforgood/casa/tree/main/spec/system_ 

### Screenshots please :)
image

### Feelings gif (optional)
_What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:_
`alt text`

### CODE REVIEW

File: app/controllers/casa_cases_controller.rb [hunk 1]
```diff
@@ -54,6 +54,8 @@ def create
         format.json { render json: @casa_case, status: :created }
       end
     else
+      set_contact_types
+      @empty_court_date = params[:casa_case][:empty_court_date] == "1"
```
Comment (elasticspoon) [inline app/controllers/casa_cases_controller.rb:58]: You can use the helper `court_date_unknown?` instead.

**Review by elasticspoon — CHANGES_REQUESTED** [review]
We have tests for this view https://github.com/rubyforgood/casa/blob/d539d06d8b3e1a07dec148be270ff7c1403e2d84/spec/system/casa_cases/new_spec.rb

I would love if you could add the small modification you made to one of the specs that checks for errors to ensure we don't have a regression in the future :slightly_smiling_face:

**Comment by iamronakgupta** [comment]
> We have tests for this view https://github.com/rubyforgood/casa/blob/d539d06d8b3e1a07dec148be270ff7c1403e2d84/spec/system/casa_cases/new_spec.rb
> 
> I would love if you could add the small modification you made to one of the specs that checks for errors to ensure we don't have a regression in the future 🙂

I did some modification, please review

**Comment by elasticspoon** [comment]
@iamronakgupta 1 last small thing. If I set the `@empty_court_date` via checkbox and there is an error the value for the check is being preserved BUT the date picker still shows. 

image

The checkbox being ticked should hide the date picker.

**Review by elasticspoon — APPROVED** [review]
LGTM!

**Review by compwron — APPROVED** [review]

**Pull request merged**